lib/mongo/retryable.rb

```diff
@@ -46,18 +46,19 @@ module Retryable
     # @return [ Result ] The result of the operation.
     #
     # @since 2.1.0
-    def read_with_retry(attempt = 0, &block)
+    def read_with_retry(attempt = 0, max_read_retries = nil, &block)
+      max_retries = max_read_retries || cluster.max_read_retries
       begin
         block.call
       rescue Error::SocketError, Error::SocketTimeoutError
         retry_operation(&block)
       rescue Error::OperationFailure => e
         if cluster.sharded? && e.retryable?
```

Comment:
This would also allow me to use the #read_with_one_retry here instead of jumping through hoops to limit the retry to 1 in the existing #read_with_retry method
https://github.com/mongodb/mongo-ruby-driver/pull/767/files#diff-fcdb9df78585e64c411303e35812189cR171